### PR TITLE
Add no-state option, filter queries from links, fix headers

### PR DESCRIPTION
**Pull request by godylockz**
Small changes from discussion.
- Fixed headers being send as `user-agent:` to `User-Agent`
- Filtering queries/anchors from extracted links
- Added new `no-state` parameter w/ shell completions updated

Tested with, no changes:
`cargo fmt; cargo clippy --all-targets --all-features -- -D warnings -A clippy::mutex-atomic; cargo test;`

### CODE REVIEW

File: src/extractor/container.rs [hunk 1]
```diff
@@ -240,8 +240,18 @@ impl<'a> Extractor<'a> {
         log::trace!("enter: get_sub_paths_from_path({})", path);
         let mut paths = vec![];
 
+        // trim whitespace, remove slashes, and queries/anchors (i.e. ?C=D;O=A)
+        let mut path_str = path.to_owned();
```
Comment (epi052) [inline src/extractor/container.rs:244]: see overall comment for proposed change

**Review by epi052 — CHANGES_REQUESTED** [review]
snippets from [rfc-3986](https://datatracker.ietf.org/doc/html/rfc3986):
```
         foo://example.com:8042/over/there?name=ferret#nose
         \_/   \______________/\_________/ \_________/ \__/
          |           |            |            |        |
       scheme     authority       path        query   fragment
```
```
The path component is terminated
   by the first question mark ("?") or number sign ("#") character, or
   by the end of the URI.
```
```
The query component is indicated by the first question
   mark ("?") character and terminated by a number sign ("#") character
   or by the end of the URI.
```

That's from the RFC for URI's. If we accept that as true, we can simplify the logic for removing the query and anchor/fragment.

```rust
    let (query_less, _discarded) = s
        .split_once('?')
        // if there isn't a '?', try to remove an anchor
        .unwrap_or_else(|| {
            // if there isn't a '#', return (original, empty)
            s.split_once('#').unwrap_or((&s, ""))
        });
```

I'd prefer the logic split out into its own function and tested directly with url paths. 

There's an even nicer solution, but it only works for 2 of the 3 calls to `add_all_sub_paths` :cry: 
```rust
                    absolute.set_query(None);

                    if self.add_all_sub_paths(absolute.path(), &mut links).is_err() {
                       ...
```

**Comment by epi052** [comment]
Looked at the changes this morning, seems to address what we spoke about. Is this ready for final review/merge?

**Comment by godylockz** [comment]
> Looked at the changes this morning, seems to address what we spoke about. Is this ready for final review/merge?

Yes I believe everything is ready. 👍

**Comment by epi052** [comment]
@godylockz whenever you get a moment, can you update this PR to point to the `release-2.6.0-multi-feature` branch instead of main please? 

Thank!

**Comment by godylockz** [comment]
All set epi!

**Comment by epi052** [comment]
awesome, thank you!

**Pull request merged**